src/base/bittorrent/torrentinfo.cpp

```diff
@@ -213,24 +214,79 @@ QByteArray TorrentInfo::metadata() const
 
 QStringList TorrentInfo::filesForPiece(int pieceIndex) const
 {
-    if (pieceIndex < 0)
-        return QStringList();
+    // no checks here since fileIndicesForPiece() will return an empty list
+    QList<int> fileIndices = fileIndicesForPiece(pieceIndex);
+
+    QStringList res;
+    std::transform(fileIndices.begin(), fileIndices.end(), std::back_inserter(res), [this](int i) {
+        return this->filePath(i);
+    });
+
```

Comment:
Are we inventing new ways to use every possible feature of the C++ language in qbittorrent? I would do it in a for loop here.
PS: I don't want you to change it. Just a remark.